x-pack/plugins/file_upload/public/importer/geojson_importer/geojson_importer.ts

```diff
@@ -18,161 +17,269 @@ import {
+          ? [ES_FIELD_TYPES.GEO_POINT, ES_FIELD_TYPES.GEO_SHAPE]
+          : [ES_FIELD_TYPES.GEO_SHAPE],
+    };
   }
 
-  public getDocs() {
-    return this._docArray;
+  public setGeoFieldType(geoFieldType: ES_FIELD_TYPES.GEO_POINT | ES_FIELD_TYPES.GEO_SHAPE) {
+    this._geoFieldType = geoFieldType;
   }
 
-  public setDocs(
-    featureCollection: FeatureCollection,
-    geoFieldType: ES_FIELD_TYPES.GEO_POINT | ES_FIELD_TYPES.GEO_SHAPE
-  ) {
-    this._docArray = [];
-    for (let i = 0; i < featureCollection.features.length; i++) {
-      const feature = featureCollection.features[i];
-      const geometry = feature.geometry as
-        | Point
-        | MultiPoint
-        | LineString
-        | MultiLineString
-        | Polygon
-        | MultiPolygon;
-      const coordinates =
-        geoFieldType === ES_FIELD_TYPES.GEO_SHAPE
-          ? {
-              type: geometry.type.toLowerCase(),
-              coordinates: geometry.coordinates,
-            }
-          : geometry.coordinates;
-      const properties = feature.properties ? feature.properties : {};
-      this._docArray.push({
-        coordinates,
-        ...properties,
-      });
+  public async import(
+    id: string,
+    index: string,
+    pipelineId: string,
+    setImportProgress: (progress: number) => void
+  ): Promise<ImportResults> {
+    if (!id || !index) {
+      return {
+        success: false,
+        error: i18n.translate('xpack.fileUpload.import.noIdOrIndexSuppliedErrorMessage', {
+          defaultMessage: 'no ID or index supplied',
+        }),
+      };
     }
+
+    const ingestPipeline = {
+      id: pipelineId,
+    };
+
+    let success = true;
+    const failures: ImportFailure[] = [...this._invalidFeatures];
+    let error;
+
+    while (this._features.length > 0 || (this._hasNext && this._isActive)) {
+      await this._readUntil(undefined, 10 * MB);
+      if (!this._isActive) {
+        return {
+          success: false,
+          failures,
+          docCount: this._totalFeatures,
+        };
+      }
+
+      let retries = IMPORT_RETRIES;
+      let resp: ImportResponse = {
+        success: false,
+        failures: [],
+        docCount: 0,
+        id: '',
+        index: '',
+        pipelineId: '',
+      };
+      const data = toEsDocs(this._features, this._geoFieldType);
+      const progress = Math.round((this._totalBytesProcessed / this._file.size) * 100);
+      this._features = [];
+      this._unimportedBytesProcessed = 0;
+
+      while (resp.success === false && retries > 0) {
+        try {
+          resp = await callImportRoute({
+            id,
+            index,
+            data,
+            settings: {},
+            mappings: {},
+            ingestPipeline,
+          });
+
+          if (retries < IMPORT_RETRIES) {
+            // eslint-disable-next-line no-console
+            console.log(`Retrying import ${IMPORT_RETRIES - retries}`);
```

Comment:
I don't think the user needs to be notified, since the REST call is retried. This is code originally taken from https://github.com/elastic/kibana/blob/7.12/x-pack/plugins/ml/public/application/datavisualizer/file_based/components/import_view/importer/importer.ts#L167 so I think its fine as is.